File: Data Structures & Algorithms/same-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def isSameTree(self, p: Optional[TreeNode], q: Optional[TreeNode]) -> bool:

        def Helper(a, b):
            if not a and not b:
                return True
            if not a or not b:
                return False
            if a.val != b.val:
                return False
            return Helper(a.left, b.left) and Helper(a.right, b.right)


        return Helper(p, q)

File: Data Structures & Algorithms/binary-tree-postorder-traversal/submission-0.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    /**
     * @param {TreeNode} root
     * @return {number[]}
     */
    postorderTraversal(root) {

        if (!root){
            return [];
        }

        let post = [];

        function helper(node){
            if (!node){
                return;
            }
            helper(node.left);
            helper(node.right);
            post.push(node.val);
        }

        helper(root);
        return post;
    }
}
